### PR TITLE
Remove `html5shiv`

### LINKED ISSUE #1477: Remove `html5shiv`
Hi all. First, thank you for the theme. It's great!

I would like to propose the removal of the `html5shiv` library. I've seen it discussed in past when `Modernizr` was removed but `html5shiv` left to support IE. I see however from the [supported browsers page](https://sphinx-rtd-theme.readthedocs.io/en/stable/development.html#supported-browsers) that IE < 9 is no longer supported.

Further, `html5shiv` is a bit outdated, not having updates & it appears no development for a few years.

Lasty, there appears to be a vulnerability reported by some scanning tools. This was reported by one of our users from a JFrog scan:
```
html5shiv html5shiv-printshiv.js ownerDocument.createDocumentFragment() Function Arbitrary Code Execution		High	security	JFrog	html5shiv	3.4 ‚â§ Version ‚â§ 3.7.3		2021-05-20T19:55:08Z	html5shiv	9.3/CVSS:2.0/AV:N/AC:M/Au:N/C:C/I:C/A:C			XRAY-78889	FALSE	npm://html5shiv:3.7.3	npm://html5shiv	sha256__d6c54436e60bc608283ddf62185350ab36b1596a0a9f27758e8238ca86424b91.tar.gz/app/wwwprivate/help/_static/js/html5shiv.min.js	unknown
html5shiv html5shiv.js ownerDocument.createDocumentFragment() Function Arbitrary Code Execution		High	security	JFrog	html5shiv	3.4 ‚â§ Version ‚â§ 3.7.3		2021-05-20T19:55:08Z	html5shiv	9.3/CVSS:2.0/AV:N/AC:M/Au:N/C:C/I:C/A:C			XRAY-78890	FALSE	npm://html5shiv:3.7.3	npm://html5shiv	sha256__d6c54436e60bc608283ddf62185350ab36b1596a0a9f27758e8238ca86424b91.tar.gz/app/wwwprivate/help/_static/js/html5shiv.min.js	unknown
```
& there is also an [open issue](https://github.com/aFarkas/html5shiv/issues/202) from SonarQube

Is it possible to remove this library? And if not, is there a way for me to exclude it during build? (atm I'm just trying to delete the lib files) Thanks!

### PR DESCRIPTION
**Pull request by mix4242**
Resolves #1477

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -263,13 +263,15 @@
       "version": "1.0.1",
       "resolved": "https://registry.npmjs.org/ajv-errors/-/ajv-errors-1.0.1.tgz",
       "integrity": "sha512-DCRfO/4nQ+89p/RK43i8Ezd41EqdGIU4ld7nGF8OQ14oc/we5rEntLCUa7+jrn3nn83BosfwZA0wb4pon2o8iQ==",
-      "dev": true
+      "dev": true,
```
Comment (mix4242) [inline package-lock.json:266]: Some changes to this file were merely caused by running `npm install --lockfile-version 1`. The full diff can be seen [here](https://github.com/readthedocs/sphinx_rtd_theme/pull/1482/commits/905a1ae0b96a99e7e54a522eb55847c6c9596bbb). Lmk if this needs changing/reverting

**Comment by mix4242** [comment]
Emily, I think it's better to voice your support and want on the [issue page](https://github.com/readthedocs/sphinx_rtd_theme/issues/1477) than on this PR itself, there are more participants there that will get comment notifications :)

**Comment by neongreen** [comment]
Unfortunately the support has withered now that I switched to Furo :/

**Comment by humitos** [comment]
@agjohnson I'd like to include this PR in 2.0 final since it's related with the drop of pretty old browsers support we mentioned there. It looks like a pretty simple change.

**Review by agjohnson — APPROVED** [review]
This change is safe, we've been in IE deprecation for a while.

However, this is another PR to probably hold off on merging until we finalize 2.0rc1, and we can treat 2.1 or 2.2 as the release to clean up more of these dependencies. Either way this should be low impact though.

**Comment by humitos** [comment]
This PR seems to be ready to merge now that we have released 2.x 🎉

**Comment by agjohnson** [comment]
Yup, this should be safe to merge now

**Review by HRYLX1 — COMMENTED** [review]
Ok

File: package-lock.json (hunk 1 above)
Comment (HRYLX1) [reply package-lock.json:266]: 已完成

**Pull request merged**